File: base/src/main/java/com/windscribe/vpn/serverlist/dao/StaticRegionDao.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.vpn.serverlist.dao;

import static androidx.room.OnConflictStrategy.REPLACE;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.windscribe.vpn.serverlist.entity.StaticRegion;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.CompletableSource;
import io.reactivex.Flowable;
import io.reactivex.Single;

@Dao
abstract public class StaticRegionDao {

    public Completable addStaticRegions(List<StaticRegion> staticRegions) {
        return deleteAll().andThen(insert(staticRegions));
    }

    @Query("Select * from StaticRegion")
    public abstract Single<List<StaticRegion>> getAllStaticRegions();

    @Query("Select * from StaticRegion")
    public abstract Flowable<List<StaticRegion>> getAllStaticRegionsFlowAble();

    @Query("Select * from staticregion where id =:id")
    public abstract Single<StaticRegion> getStaticRegionByID(int id);

    @Query("SELECT Count(*) FROM staticregion")
    public abstract Single<Integer> getStaticRegionCount();

    @Query("Delete from StaticRegion")
    abstract Completable deleteAll();

    @Insert(onConflict = REPLACE)
    abstract CompletableSource insert(List<StaticRegion> staticRegions);

    @Query("Delete from StaticRegion")
    public abstract void clean();
}

File: base/src/main/java/com/windscribe/vpn/serverlist/dao/StaticRegionDao.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.vpn.serverlist.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy.Companion.REPLACE
import androidx.room.Query
import com.windscribe.vpn.serverlist.entity.StaticRegion

@Dao
abstract class StaticRegionDao {

    suspend fun addStaticRegions(staticRegions: List<StaticRegion>) {
        deleteAll()
        insert(staticRegions)
    }

    @Query("Select * from StaticRegion")
    abstract suspend fun getAllStaticRegions(): List<StaticRegion>

    @Query("Select * from staticregion where id =:id")
    abstract suspend fun getStaticRegionByIDAsync(id: Int): StaticRegion?

    @Query("Delete from StaticRegion")
    abstract suspend fun deleteAll()

    @Insert(onConflict = REPLACE)
    abstract suspend fun insert(staticRegions: List<StaticRegion>)

    @Query("Delete from StaticRegion")
    abstract fun clean()
}